vmdb/app/models/configuration_manager_foreman.rb

```diff
@@ -1,7 +1,19 @@
 class ConfigurationManagerForeman < ConfigurationManager
-  delegate :connection_attrs, :name, :to => :provider
+  delegate :raw_connect, :connection_attrs, :name, :to => :provider
 
   def self.ems_type
     "foreman_configuration".freeze
   end
+
+  def with_provider_connection(options = {})
+    raise "no block given" unless block_given?
+    log_header = "MIQ(#{self.class.name}.with_provider_connection)"
```

Comment:
Can the ```log_header``` be moved into the ```$log.info``` line since that is the only place it's being used?  Also, it could be shortened to ```"MIQ(#{self.class.name}##{__method__})"```